extensions/csrf-reactive/runtime/src/main/java/io/quarkus/csrf/reactive/runtime/CsrfRequestResponseReactiveFilter.java

```diff
@@ -87,12 +87,26 @@ public void filter(ResteasyReactiveContainerRequestContext requestContext, Routi
         if (requestMethodIsSafe(requestContext)) {
             // safe HTTP method, tolerate the absence of a token
             if (isCsrfTokenRequired(routing, config)) {
-                // Set the CSRF cookie with a randomly generated value
-                byte[] tokenBytes = new byte[config.tokenSize];
-                secureRandom.nextBytes(tokenBytes);
-                routing.put(CSRF_TOKEN_BYTES_KEY, tokenBytes);
-                routing.put(CSRF_TOKEN_KEY, Base64.getUrlEncoder().withoutPadding().encodeToString(tokenBytes));
-
+                if (cookieToken == null) {
+                    generateNewCsrfToken(routing, config);
+                } else {
+                    String csrfTokenHeaderParam = requestContext.getHeaderString(config.tokenHeaderName);
+                    if (csrfTokenHeaderParam != null) {
+                        LOG.debugf("CSRF token found in the token header");
+                        // Verify the header, make sure the header value, possibly signed, is returned as the next cookie value
+                        verifyCsrfToken(requestContext, routing, config, cookieToken, csrfTokenHeaderParam);
+                    } else if (!config.tokenSignatureKey.isEmpty()) {
+                        // If the signature is required, then we can not use the current cookie value
+                        // as the HTML form token key because it represents a signed value of the previous key
+                        // and it will lead to the double-signing issue if this value is reused as the key.
+                        // It should be fine for simple HTML forms anyway
```

Comment:
I do not understand this comment. We have two ways to get a token: cookie or header. Here we're in a situation where we have a signed cookie, and we don't have a header. Why do we have to regenerate a token? Why would the previous signature be invalid?

Reply:
I guess this would lead to double-signing given that signing is done in the ResponseFilter. 

Is there a specific reason why signing cannot be done in generateNewCsrfToken(..) and use a single routing parameter?